normandy/recipes/management/commands/update_addon_urls.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from urllib.parse import urlparse, urlunparse
+from django.core.management.base import BaseCommand
+
+from normandy.recipes.models import RecipeRevision
+
+
+class Command(BaseCommand):
+    """
+    Rewrite all revisions to update the URL of add-ons stored in Normandy's
+    file storage to a new hostname.
+    """
+
+    help = "Updates add-on URL in revisions"
+
+    def add_arguments(self, parser):
+        parser.add_argument("new_hostname", help="Hostname to convert add-on urls to")
+
+    def handle(self, *args, new_hostname, **options):
+        target_revisions = RecipeRevision.objects.filter(action__name="opt-out-study")
+        update_count = 0
+        for rev in target_revisions:
+            if "addonUrl" not in rev.arguments:
```

Comment:
If you have `arguments = {'addonUrl': None}` this line wouldn't notice but later it would fail on `urlparse(None)`.

Does that matter?

Reply:
That's a good point. I'll add a check for that (as improbably as it is).